src/parser/jobs/ast/modules/Divination.tsx

```diff
@@ -0,0 +1,297 @@
+
+				playersBuffed: [],
+				overlap: false,
+			}
+			this.castHook = this.addEventHook(
+				filter<Event>()
+					.source(this.parser.actor.id)
+					.type('action'),
+				this.onCast,
+			)
+
+			//to set other ast information if active and overwritten
+			//note assumption that overwritten happens if duration hasn't passed
+			if (this.data.statuses.DIVINATION.duration > this.currentWindow.start - this.otherAst.start) {
+				this.otherAst.countOverwritten++
+				this.otherAst.timeOverwritten += (this.otherAst.start + this.data.statuses.DIVINATION.duration) - event.timestamp
+				this.currentWindow.overlap = true
+			}
+			this.otherAst.active = false
+			this.otherAst.start = 0
+		}
+	}
+
+	private tryCloseWindow(event: Events['statusRemove']) {
+		//note: closing other AST window handled in open window since closing the window doesn't catch what we need because status remove happens before status apply
+
+		//to stop using function if window isn't even open. used when splicing
+		if (this.currentWindow == null) {
+			return
+		}
+
+		//close window and push to history
+		if (event.source === this.parser.actor.id) {
+			this.currentWindow.end = event.timestamp
+			this.history.push(this.currentWindow)
+			this.currentWindow = undefined
+			if (this.castHook != null) {
+				this.removeEventHook(this.castHook)
+				this.castHook = undefined
+			}
+		}
+	}
+
+	private onCast(event: Events['action']) {
+
+		// If we don't have a window, bail
+		if (this.currentWindow == null) {
+			return
+		}
+
+		const action = this.data.getAction(event.action)
+
+		// Can't do anything else if we didn't get a valid action object
+		if (action == null) {
+			return
+		}
+
+		// Add the action to the list
+		this.currentWindow.rotation.push(event)
+
+		if (action.onGcd) {
+			this.currentWindow.gcdCount++
+
+			// If we haven't recorded a trailing GCD event for this closed window, do so now
+			if (this.currentWindow.end != null && this.currentWindow.trailingGcdEvent == null) {
+				this.currentWindow.trailingGcdEvent = event
+			}
+		}
+	}
+
+	private onComplete() {
+		if (this.otherAst.countOverwritten !== 0) {
+			/*
+				SUGGESTION: Overwriting other AST's divination
+			*/
+			this.suggestions.add(new Suggestion({
+				icon: this.data.actions.DIVINATION.icon,
+				content: <Trans id="ast.divination.suggestion.usage.content">
+					Consider coordinating with your co-Astrologian to maximize the amount of time <DataLink action="DIVINATION" /> is up so the party can benefit from extra damage for longer.
+				</Trans>,
+				severity: SEVERITY.MAJOR,
+				why: <Trans id="ast.divination.suggestion.usage.why">
+					About {this.otherAst.countOverwritten} <Plural value={this.otherAst.countOverwritten} one="cast" other="casts" /> of <DataLink action="DIVINATION" /> were overwritten resulting in a loss of at least {this.parser.formatDuration(this.otherAst.timeOverwritten)}.
+				</Trans>,
+			}))
+		}
+	}
+
+	// just output, no suggestions for now.
```

Comment:
Comment is incorrect, as we are generating suggestions.

Reply:
removed oop